libnmstate/nm/device.py

```diff
@@ -249,11 +249,6 @@ def delete_device(nmdev):
 
 def _safe_delete_device_async(nmdev):
     mainloop = nmclient.mainloop()
-    if nmdev.get_state() == nmclient.NM.DeviceState.DEACTIVATING:
```

Comment:
I think the commit explains it, but it may be phrased in a way which may be confusing.
Although the state of the device is DEACTIVATING, no new event is returned to finalize and confirm the device removal. Therefore, although we expect the device to be removed, we cannot trust it and explicitly perform a device removal.

Reply:
I have updated the commit message with my foundings, please review.

Reply:
I have a little bit of trouble to understand the first paragraph, is this correct?

```
The current code skips device deletion when the device is in DEACTIVATING
state. This can result in leftover software interfaces without corresponding
Network manager profile. Therefore try to remove the device regardless of its state.

A proper fix would be to wait for Network Manager to delete it and only do it explicitly after a timeout. However,
when `NM.Device` is in DEACTIVATING state, after registering both
`NM.Device.state-change` and `NM.Client.device-removed` signals,
no callback was triggered by NetworkManager-1.20.9-1.37c571a72a

Jira issue to  find a better solution: https://nmstate.atlassian.net/browse/NMSTATE-280
```

But I am wondering, if this solution works, why do we need a better solution?

Reply:
I guess better solution means a solution without explanation: if in DEACTIVATING, we wait on it, if timeout, we delete explicitly.

Reply:
The new code will just always delete. It does not need an explanation by itself, only why removing the old exception is better needs an explanation. Since removing the exception does not seem tohurt, it makes me wonder why we had it there in the first place.